stl/inc/ranges

```diff
@@ -3603,14 +3605,28 @@ namespace ranges {
         };
 
         template <bool _Const>
-        class _Iterator
+        class _Iterator_base
+            : public _Category_base<_Maybe_const<_Const, _Vw>, _InnerRng<_Const>, is_reference_v<_InnerRng<_Const>>> {};
+
+        template <bool _Const>
+            requires forward_range<_Vw>
+        class _Iterator_base<_Const>
             : public _Category_base<_Maybe_const<_Const, _Vw>, _InnerRng<_Const>, is_reference_v<_InnerRng<_Const>>> {
+        protected:
+            using _OuterIter = iterator_t<_Maybe_const<_Const, _Vw>>;
+
+            _Iterator_base() = default;
+            constexpr explicit _Iterator_base(_OuterIter&& _Outer_) : _Outer{_STD move(_Outer_)} {}
+
+            /* [[no_unique_address]] */ _OuterIter _Outer{};
+        };
+
+        template <bool _Const>
+        class _Iterator : public _Iterator_base<_Const> {
         private:
-            template <bool>
-            friend class _Iterator;
-            template <bool>
-            friend class _Sentinel;
+            friend join_view;

```

Comment:
This is great (it aligns with what we've been doing in the other views). Suggestion: Should this change also be made to `_Sentinel`, for consistency within this view?